### DISCUSSION TITLE
Unit testing

### DISCUSSION THREAD

**Discussion by digitall-it** [spatie/laravel-data#871] [Q&A]
The documentation is missing a clear example on how to unit test a laravel-data class. As the package relies on config facade, it cannot resolve the dependency in the unit tests, and the user has to figure out himself what to do or move the test to feature tests so it will have to boot the entire Laravel application. I would like to find a golden standard solution so that I can try making a PR to the documentation. For example, let's imagine the SongData structure used as an example in the documentation. You may argue that DTO are simple objects that don't need any form of testing, and that could be a valid point per se, but it could be worth a shot to cover all the bases. Thank you!

**Comment by hogeorge** [comment]
I am also facing this issue, is there a documentation on how to unit test Spatie magic creation methods in unit tests where the app is not bootstrapped (test classes extend PHPUnit\Framework\TestCase)

**Reply by digitall-it** [reply]
My solution so far for any solution that requires bootstrapping is to refactor it as a Feature test.

**Comment by igorleszczynski** [comment]
Same problem here

**Comment by igorleszczynski** [comment]
@digitall-it 
@hogeorge 
Look at these comment
https://github.com/spatie/laravel-data/discussions/178#discussioncomment-6116074
